nomenclature/cli.py

```diff
@@ -41,10 +43,34 @@ def cli_valid_yaml(path: Path):
     default="definitions",
 )
 def cli_valid_project(
-    path: Path, dimensions: List[str], mappings: str, definitions: str
+    path: Path,
+    dimensions: Optional[List[str]],
+    mappings: Optional[str],
+    definitions: str,
```

Comment:
```suggestion
    definitions: Optional[str],
```

Reply:
I don't think `Optional` would be correct here. In the python typing sense `Optional` specifies a parameter that can be `None` not necessarily an optional parameter (meaning there is a default value). Here, the function would fail if `None` was used as an input.

Reply:
Ok, I see your point but I find it confusing that a "required" (or non-optional) argument comes after optional ones. I suggest to either change the order (for both the CLI and the `testing.assert_valid_structure()`) or set the default for *definitions* also to `None` (and replace it in the code).

Reply:
Fair point, I would vote for changing the order then. When invoking the cli with the parameter names the order does not matter anyway. I would leave the string default in the function definition since it's more concise.

Reply:
Fine to me. We loose the order that @danielhuppmann recommended last week with 'dimensions' first because it will be used more, but OK to me.